modules/git/git.go

```diff
@@ -180,6 +181,24 @@ func InitOnceWithSync(ctx context.Context) (err error) {
 			return
 		}
 
+		// Gitea 1.17-rc uses "setting.RepoRootPath" for Git HOME, which is incorrect.
+		// Do this check to make sure there is no legacy file in the RepoRootPath. This check might be able to be removed with 1.19 release.
+		var hasCheckErr bool
+		_ = os.Remove(filepath.Join(setting.RepoRootPath, ".gitconfig")) // remove the auto generated git config file
+		_ = os.Remove(filepath.Join(setting.RepoRootPath, ".ssh"))       // remove the empty dummy ".ssh" directory
+		for _, wellKnownName := range []string{".ssh", ".gnupg"} {
+			checkLegacyFile := filepath.Join(setting.RepoRootPath, wellKnownName)
+			_, checkErr := os.Stat(checkLegacyFile)
+			if checkErr == nil || !errors.Is(checkErr, os.ErrNotExist) {
+				log.Error(`Git HOME has been moved to [git].HOME_PATH, but there are legacy file in old place. Please backup and remove the legacy files %q`, checkLegacyFile)
+				hasCheckErr = true
+			}
+		}
+		if hasCheckErr {
+			log.Fatal("Please fix errors above, remove legacy files")
+		}
+		// end of legacy Gitea 1.17-rc check
+
```

Comment:
I'm not sure we should bother doing this. There's no way we can be certain that any of these files are from us and therefore we should not touch them.

Reply:
My thought was:

* Normally, there should be no org/user name starting with dot, although the name check is not strict at the moment.
* If the `.gnupg` was left there, there might be private keys in it, which is not ideal for production.

For the result of this logic:

* If a Gitea user doesn't have put their private key in `.gnupg` and doesn't have conflicting org/user names (.gitconfig/.ssh/.gnupg), they feel nothing, everything will be correct in the end (this case covers nearly 100% users).
* If a Gitea user tried Gitea-1.17rc and has put their private key in `.gnupg`, they must know the system very well and can fix the problem easily (only a few, we could also notify them in the related issue comment)
* If a Gitea user has conflicting org/user names (.gitconfig/.ssh/.gnupg), this is the only case that the breaking happens, IMO since the dot files are consensus for reserved use, it's better for them to avoid such user/org names. It would only affect very few users.

And this code is safe, it won't remove valuable user files, it only removes the .gitconfig and empty directories. Maybe the `.gitconfig` could be moved to the new home automatically?

Do you have more suggestions about this?